File: src/BackEnd/autossanchez_service/src/main/java/net/autossanchez/entity/Vehicle.java
package net.autossanchez.entity;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Vehicle {

	@Id
	@GeneratedValue
	private Long id;

	private String vin;
	private String number_plate;
	private String brand;
	private String model;
	private String engine;
	private int price;
	
	 @ManyToOne(fetch = FetchType.EAGER,optional = false)
	 @JoinColumn(name = "user_id",nullable = false)
	private User user_id;
}

File: src/FrontEnd/src/app/components/Vehicle/model/Vehicle.ts
import { User } from 'src/app/model/User';

export class Vehicle {
    id:number;
    vin:String;
    number_plate:String;
    brand:String;
    model:String;
    engine:String;
    price:number;
    fuel:String;
    color:String;
    extra:String;
    user_id:User;
}